### DISCUSSION TITLE
Feature Request - Port to Firefox

### DISCUSSION THREAD

**Discussion by codethought** [coddingtonbear/obsidian-web#79] [Ideas]
If you could port this over to Firefox many of us using Obsidian would be most grateful. Thank you for your consideration.

**Comment by qbit** [comment]
I have a fork that enables the extension to run in Firefox: https://github.com/qbit/obsidian-web/tree/firefox

There are a few things it needs:
- The manifest.json is different for firefox, but there is a lot of overlap, perhaps a tool like `jq` should be used to merge two JSON blobs into a browser specific manifest?
- Firefox requires an extension ID. ATM I just have a dummy placeholder.
- I am not sure the `sandbox` changes are correct - best I can tell firefox has no equiv of the `sandbox` feature in Chrome's manifest.

**Reply by SebastianAtWork** [reply]
How do I install this in Firefox? I build it via npm run build and zipped the result but firefox says the extension is corrupt when trying to install it from file.

**Reply by qbit** [reply]
Run `npm run dist-chrome` (yes, chrome, I piggybacked for now :D) and it will produce a zip file called `obsidian-2.2.3.chrome.zip`

Then in firefox go to `about:debugging#/runtime/this-firefox` and click `Load Temporary Add-on`. From there select the freshly built zip file.

Since it's temporary - you have to do that every time firefox starts.

**Reply by coddingtonbear** [reply]
I'd be somewhat surprised that things would work without sandboxing -- I'm vaguely aware that chrome's extension rules don't allow for using an iframe with `sandbox=` set, and would be a little surprised if Firefox did, too.  Do the templating features actually work in your fork there?

**Reply by qbit** [reply]
hm. I am not sure - I have been uploading existing templates. Not sure if that uses the same path.

The template I tested with calls some `{{date HH:mm}}` kind of things. They seem to work as expected.

**Reply by johannesCmayer** [reply]
For me, it says that the manifest is not valid. How do you fix this issue? I have no idea about extension development, so I'd appreciate the process being broken down into simple steps.

When I unzip the generated zip, there is no manifest inside.

**Comment by tamouse** [comment]
Has there been any more progress on this? Not nagging, just wondering if I should switch back to Chrome. 🤣

**Reply by codethought** [reply]
@tamouse  - someone did a fork if you're brave and willing... 

https://github.com/qbit/obsidian-web/tree/firefox

**Comment by sammcj** [comment]
Any updates on making this officially work in Firefox as well as Chrome?

**Comment by dundalek** [comment]
The previous port does not seem to be available, I created a new one: https://github.com/coddingtonbear/obsidian-web/pull/235
